pkg/capture/capture.go

```diff
@@ -292,17 +292,35 @@ func (c *Capture) process() <-chan error {
 				captureErrors <- fmt.Errorf("capture error: %w", err)
 				return
 			}
-			c.stats.Processed++
 
 			// Parse the packet, extract relevant data and add to the flow log
 			// Note: Since the compiler fails to inline this as a function, it is kept in the main loop
 			if iplayerType := ipLayer.Type(); iplayerType == ipLayerTypeV4 {
 				epHash, direction, errno := ParsePacketV4(ipLayer)
+
+				// Check for issues / errors during parsing (checked inline to avoid unnecessary function
+				// call to ParsePacket...())
+				if errno > capturetypes.ErrnoOK {
+					c.updateParsingErrorCounters(errno)
+					continue
+				}
+
+				c.stats.Processed++
```

Comment:
As discussed: it feels obvious to put this on top before the processing.

A comment why it isn't in this particular case would be helpful.